File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        
        closedList = [")","}","]"]
        if s[0] in closedList:
            return False
        stack = []

        for i in s:
            if i not in closedList:
                stack.append(i)
            else:
                if stack == []:
                    return False
                k = stack.pop()
                if (i == "}" and k == "{") or ( i == ")" and k == "(" ) or (i == "]" and k == "["):
                    pass
                else:
                    return False
        if stack == []:
            return True
        return False

File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        
        Stack<Character> stack = new Stack<>();
        
        for(char c : s.toCharArray()){

            if(c == ')'){
                if(stack.isEmpty() || stack.peek() != '('){
                    return false;
                }
                stack.pop();
            }
            else if(c == '}'){
                if(stack.isEmpty() || stack.peek() != '{'){
                    return false;
                }
                stack.pop();
            }
            else if(c == ']'){
                if(stack.isEmpty() || stack.peek() != '['){
                    return false;
                }
                stack.pop();
            }
            else{
                stack.push(c);
            }
            
        }
        if(!stack.isEmpty()){
            return false;
        }
        return true;
    }
}